pkg/health/condition/check_backup_ready.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package condition
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	coordinationv1 "k8s.io/api/coordination/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type backupReadyCheck struct {
+	cl client.Client
+}
+
+const (
+	FullBackupSucceeded string = "FullBackupSucceeded"
+	IncrBackupSucceeded string = "IncrementalBackupSucceeded"
+	BackupFailed        string = "BackupFailed"
+	Unknown             string = "Unknown"
+)
+
+func (a *backupReadyCheck) Check(etcd druidv1alpha1.Etcd) Result {
+	//Default case
+	result := &result{
+		conType: druidv1alpha1.ConditionTypeBackupReady,
+		status:  druidv1alpha1.ConditionFalse,
+		reason:  Unknown,
+		message: "Cannot determine etcd backup status since no snapshot leases present",
+	}
+
+	//Fetch snpashot leases
+	fullSnapLease := &coordinationv1.Lease{}
+	deltaSnapLease := &coordinationv1.Lease{}
+	var fullSnapErr, incrSnapErr error
+	fullSnapErr = a.cl.Get(context.TODO(), types.NamespacedName{Name: getFullSnapLeaseName(&etcd), Namespace: "default"}, fullSnapLease)
+	incrSnapErr = a.cl.Get(context.TODO(), types.NamespacedName{Name: getDeltaSnapLeaseName(&etcd), Namespace: "default"}, deltaSnapLease)
+
+	if fullSnapErr != nil && incrSnapErr != nil {
+		return result
+	}
+
+	//Cases where snpashots are taken and lease updated
+	if incrSnapErr == nil && deltaSnapLease.Spec.RenewTime != nil {
+		deltaLeaseRenewTime := deltaSnapLease.Spec.RenewTime
+		if time.Since(deltaLeaseRenewTime.Time) < 2*etcd.Spec.Backup.DeltaSnapshotPeriod.Duration {
+			result.reason = IncrBackupSucceeded
+			result.message = "Delta snapshot taken"
+			result.status = druidv1alpha1.ConditionTrue
+			return result
+
+		}
+	}
+	if fullSnapErr == nil && fullSnapLease.Spec.RenewTime != nil {
+		fullLeaseRenewTime := fullSnapLease.Spec.RenewTime
+		if time.Since(fullLeaseRenewTime.Time) < 2*etcd.Spec.Backup.DeltaSnapshotPeriod.Duration {
```

Comment:
Hm, full snapshot leases are only renewed after full snapshot or compaction. So the check here will very likely fail, right?

Reply:
Yes, this will be false almost always except when a full snapshot is taken. 
In that case, the delta snapshot case just before this would decide the condition.

Do you think it makes sense to combine this with the block above and just consider to be true if either snapshot is renewed with a common message instead?

Reply:
Good question. Does it make sense to check if the last full backup at least happened in the last 24h (this is the only thing we can presume)?

Reply:
On it's own, I don't think that would be a good idea. 24h would be too large of a window to consider while setting the condition to true.
If you mean a check to see if a full snap has happened in the past 24h together with the delta snap check, that would make sense

Reply:
> If you mean a check to see if a full snap has happened in the past 24h together with the delta snap check, that would make sense

Yep, that's what I meant 👍

Reply:
Yup, this makes sense, I'll add it and push a new commit